easybuild/easyblocks/l/lammps.py

```diff
@@ -228,11 +228,91 @@ def update_kokkos_cpu_mapping(self):
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Aug2023')):
             self.kokkos_cpu_mapping['icelake'] = 'ICX'
             self.kokkos_cpu_mapping['sapphirerapids'] = 'SPR'
+            self.kokkos_cpu_mapping['skylake'] = 'SKL'
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('4Feb2025')):
             self.kokkos_cpu_mapping['neoverse_v2'] = 'ARMv9-Grace'
         if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Apr2025')):
             self.kokkos_cpu_mapping['zen4'] = 'ZEN4'
 
+    def get_kokkos_arch(self, cuda_cc, kokkos_arch):
+        """
+        Return KOKKOS ARCH in LAMMPS required format, which is 'CPU_ARCH' and 'GPU_ARCH'.
+
+        see: https://docs.lammps.org/Build_extras.html#kokkos
+        """
+        # CPU arch
+        # NOTE: if the CPU KOKKOS_ARCH flag is specified, Kokkos will add the correspondent `-march` and `-mtune` flags
+        # to the compiler flags, which may override the ones set by EasyBuild.
+        # https://github.com/lammps/lammps/blob/stable_29Aug2024/lib/kokkos/cmake/kokkos_arch.cmake#L228-L531
+        processor_arch = None
+        if build_option('optarch') == OPTARCH_GENERIC:
```

Comment:
```suggestion
        if build_option('optarch') == GENERIC:
```
I'm not that familiar with doing this but it seems that is `GENERIC` and not `OPTARCH_GENERIC`.